core/src/main/java/org/carbondata/core/cache/dictionary/AbstractColumnDictionaryInfo.java

```diff
@@ -241,22 +257,21 @@ private void decrementAccessCount() {
    */
   protected byte[] getDictionaryBytesFromSurrogate(int surrogateKey) {
     byte[] dictionaryValueInBytes = null;
-    int totalSizeOfDictionaryChunksTraversed = 0;
-    for (List<byte[]> oneDictionaryChunk : dictionaryChunks) {
-      totalSizeOfDictionaryChunksTraversed =
-          totalSizeOfDictionaryChunksTraversed + oneDictionaryChunk.size();
-      // skip the dictionary chunk till surrogate key is lesser than size of
-      // dictionary chunks traversed
-      if (totalSizeOfDictionaryChunksTraversed < surrogateKey) {
-        continue;
+    if (surrogateKey > 0) {
+      // surrogate key starts from 1 and list index will start from 0, so lets say if surrogate
+      // key is 10 then value will present at index 9 of the dictionary chunk list
+      int actualSurrogateIndex = surrogateKey - 1;
+      // lets say dictionaryOneChunkSize = 10, surrogateKey = 10, so bucket index will
+      // be 0 and dictionary chunk index will be 9 to get the value
+      int dictionaryBucketIndex = actualSurrogateIndex / dictionaryOneChunkSize;
+      if (dictionaryChunks.size() > dictionaryBucketIndex) {
```

Comment:
This if check is required?? As i cannot find any scenario when dictionary bucket index will more than dicitonaryChunkSize.